tmt/steps/execute/__init__.py

```diff
@@ -1129,9 +1129,11 @@ def go(self, force: bool = False) -> None:
         """ Execute tests """
         super().go(force=force)
 
-        # Clean up possible old results
+        # Clean up possible old results, by switching them to pending state
         if force:
-            self._results.clear()
+            for index, result in enumerate(self._results):
+                if result.result != ResultOutcome.PENDING:
+                    self._results[index] = Result(name=result.name, result=ResultOutcome.PENDING)
```

Comment:
I'd say this should recreate the list from scratch, i.e. not overwriting the existing results. Let's say there was a test with custom results, creating 100 of results for that single test. Before running the test for the first time, there would be a single `PENDING` placeholder for the test; after the operation above, `results.yaml` may contain 100 results for that test, each with `PENDING` results. And since tmt does not control creation of these results, the test may easily decide to not produce several of them this time, and therefore that particular result should never appear in our `results.yaml`. Dropping all results except the single `PENDING` one per test seems like better option to me.